File: src/lessons/welcome/methods/picture/PictureMono3Entity.scala
package lessons.welcome.methods.picture;

import plm.universe.bugglequest.SimpleBuggle;

class PictureMono3Entity extends SimpleBuggle {

	/* BEGIN TEMPLATE */
	override def run() {
		/* BEGIN SOLUTION */
		for (i <- 1 to 9) {
			makeLine(9);
			nextLine();
		}
	}
	def mark() {
		brushDown();
		brushUp();
	}

	def makeV() {
		forward(2);
		mark();

		stepForward();
		left();
		stepForward();
		mark();

		stepBackward();
		right();
		stepForward();
		mark();

		forward(2);
		left();
	}

	def makePattern() {
		makeV();
		makeV();
		makeV();
		makeV();
		forward(7);
	}

	def makeLine(count: Int){
		for (i <- 1 to count)
			makePattern();
		backward(count*7);
	}

	def nextLine() {
		left();
		forward(7);
		right();	
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/methods/picture/PictureMono3Entity.py
# BEGIN SOLUTION
def mark():
   brushDown()
   brushUp()

def makeV():
   forward(2)
   mark()
   forward()
   left()
   forward()
   mark()
   backward()
   right()
   forward()
   mark()   
   forward(2)
   left()

def makePattern():
    for i in range(4):
        makeV()
    forward(7)

def makeLine(count):
   for i in range(count):
      makePattern()
   backward(count*7)

def nextLine():
   left()
   forward(7)
   right()

for i in range(9):
   makeLine(9)
   nextLine()
# END SOLUTION
